### DISCUSSION TITLE
Implement 5% Withdrawal Fee for Creator's Pre-OSS Token Fee

### DISCUSSION THREAD

**Discussion by defi-er** [dev-protocol/DIPs#35] [Ideas]
Implement a 5% withdrawal fee for creators that joined pre OSS token fee to level the playing ground for creators. The DEV from the 5% fee should be sent to the Treasury created for OSS tokens.

**Comment by serjames1** [comment]
All for it

**Comment by funkyduckerino** [comment]
I totally agree with that

**Comment by sirdesu** [comment]
Great idea!

**Comment by MidgetGems** [comment]
Seems a Sensible suggestion to me. The staker community seem to support the idea would be interesting to hear from the creator side.

**Comment by aggre** [comment]
I can think of this proposal as a proposal for fairness.

In my idea, whether the protocol collects withdrawal fees depends on whether the target creator token contributes to Treasury. In other words, any creator can voluntarily deposit tokens in Treasury, so if the creator deposits tokens, the creator will not be charged withdrawal fees from the protocol. And technically, the protocol can't tell when a creator token was issued, so I think it's a good idea to decide it based on Treasury balance.

My idea is to update Treasury to support voluntary deposits by creators. It is already possible to deposit any balance in Treasury, but I think the deposit should always be 5%. (The rate is determined by the Policy, so the value of "5%" will fluctuate.)

**Reply by Cold-A-Muse** [reply]
Would this be something that could be turned on/off through stakes.social?

**Reply by aggre** [reply]
"No" in my opinion. This is because the treasury does not support user-transacted refunds. Conversely, if stakes.social supports "on/off" by the user, the treasury must support user-transacted refunds.

**Reply by Cold-A-Muse** [reply]
> "No" in my opinion. This is because the treasury does not support user-transacted refunds. Conversely, if stakes.social supports "on/off" by the user, the treasury must support user-transacted refunds.

Why would a creator voluntarily deposit tokens into the treasury? Would there be an incentive to do so?

**Reply by defi-er** [reply]
" And technically, the protocol can't tell when a creator token was issued, so I think it's a good idea to decide it based on Treasury balance."

@aggre, what if the logic is the following:

If the Creator token withdrawing rewards doesn't have 5% of its supply in the Dev Protocol treasury then charge a 5% fee

**Considerations**
In the future if Dev Protocol Treasury moves 5% supply then logic is incorrect therefore it should incorporate "**_if the smart treasury ever received_**"

**Reply by aggre** [reply]
> Why would a creator voluntarily deposit tokens into the treasury? Would there be an incentive to do so?

@defi-er For example, It can remove the 5% fee for all holders. An author who wants a lot of committers or stakeholders can reduce their fees by depositing 5% own.

**Comment by aggre** [comment]
@defi-er What specs do you consider ideal in the following cases?

- When the 5% fee changes to **6%**, the handling method of fees for users who deposited 5%
- When the 5% fee changes to **4%**, the handling method of fees for users who deposited 5%

**Reply by defi-er** [reply]
I don't understand the cases, could you please rephrase them?

**Reply by aggre** [reply]
I paraphrase them as follows:

1. Alice created the Property when the Treasury fee was 5%. When the Treasury fee is changed to 6%, how will Alice be affected?
2. Alice created the Property when the Treasury fee was 5%. When the Treasury fee is changed to 4%, how will Alice be affected?
